File: postman/sdks/Specs SDKs/Search Service/python/src/search_service_sdk/models/multisearch_response.py
from __future__ import annotations
from pydantic import Field
from typing import Optional
from typing import Union
from .utils.base_model import BaseModel
from .tracks_response import TracksResponse
from .artists_response import ArtistsResponse
from .charts_response import ChartsResponse
from .labels_response import LabelsResponse
from .releases_response import ReleasesResponse


class MultisearchResponse(BaseModel):
    """Response model for the `all-search` endpoint.

    :param tracks: Response model for the `tracks` endpoint.
    :type tracks: TracksResponse
    :param artists: artists
    :type artists: ArtistsResponse
    :param charts: charts
    :type charts: ChartsResponse
    :param labels: labels
    :type labels: LabelsResponse
    :param releases: Response model for the `releases` endpoint.
    :type releases: ReleasesResponse
    """

    tracks: TracksResponse = Field(
        description="Response model for the `tracks` endpoint."
    )
    artists: ArtistsResponse
    charts: ChartsResponse
    labels: LabelsResponse
    releases: ReleasesResponse = Field(
        description="Response model for the `releases` endpoint."
    )

File: postman/sdks/Specs SDKs/Search Service/java/src/main/java/com/searchservicesdk/models/MultisearchResponse.java
package com.searchservicesdk.models;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.ToString;
import lombok.With;
import lombok.extern.jackson.Jacksonized;

/**
 * Response model for the `all-search` endpoint.
 */
@Data
@Builder
@With
@ToString
@EqualsAndHashCode
@Jacksonized
public class MultisearchResponse {

  /**
   * Response model for the `tracks` endpoint.
   */
  @NonNull
  private TracksResponse tracks;

  @NonNull
  private ArtistsResponse artists;

  @NonNull
  private ChartsResponse charts;

  @NonNull
  private LabelsResponse labels;

  /**
   * Response model for the `releases` endpoint.
   */
  @NonNull
  private ReleasesResponse releases;
}
